tests/legacy-cli/e2e/tests/basic/update-7.0.ts

```diff
@@ -0,0 +1,39 @@
+import { createProjectFromAsset } from '../../utils/assets';
+import { ng, silentNpm } from '../../utils/process';
+import {
+  isPrereleaseCli, removeHttpDep, useBuiltPackages, useCIChrome, useCIDefaults,
+} from '../../utils/project';
+
+
+export default async function () {
+  const extraUpdateArgs = await isPrereleaseCli() ? ['--next', '--force'] : [];
+
+  // Create new project from previous version files.
+  // We must use the original NPM packages to force a real update.
+  // This also means that we can only truly test updates after a release.
+  await createProjectFromAsset('7.0-project', true);
+  // Update the CLI.
+  await ng('update', '@angular/cli', ...extraUpdateArgs);
+  // Use the packages we are building in this commit, and CI Chrome.
+  await useBuiltPackages();
+  await useCIChrome('src/');
+  await useCIChrome('e2e/');
+  await useCIDefaults('seven-oh-project');
+  await silentNpm('install');
+  // TODO: Remove http? Should this be part of the migration? Or even fail it.
+  await removeHttpDep();
+  // Update Angular.
+  await ng('update', '@angular/core', ...extraUpdateArgs);
+  // There's a weird warning here in the step above:
+  // Cannot determine the module for class AppComponent in
+  // ..../src/app/app.component.ts! Add AppComponent to the NgModule to fix it.
+  // Cannot determine the module for class LazyCompComponent in
+  // ..../src/app/lazy/lazy-comp/lazy-comp.component.ts! Add LazyCompComponent to the NgModule to
+  // fix it.
+  // Run CLI commands.
+  await ng('generate', 'component', 'my-comp');
+  await ng('test', '--watch=false');
+  await ng('lint');
+  await ng('e2e');
```

Comment:
Blocked on https://github.com/angular/angular-cli/pull/14222, and it being released.

Reply:
Also blocked on https://github.com/angular/angular-cli/issues/14218, and it being released.